stack/core/src/main/java/org/apache/usergrid/corepersistence/CpEntityManager.java

```diff
@@ -729,6 +740,43 @@ public void createApplicationCollection( String entityType ) throws Exception {
         create( entityType, null );
     }
 
+    @Override
+    public Entity getAllEntityFromFields( String collectionType, String aliasType ){
+        String collName = Schema.defaultCollectionName( collectionType );
+
+        CollectionScope collectionScope = getCollectionScopeNameFromEntityType(
+            getApplicationScope().getApplication(), collName);
+
+        String propertyName = Schema.getDefaultSchema().aliasProperty( collName );
```

Comment:
IIRC entities can have more than 1 alias property.  You might want to verify this.